flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/casting/ArrayToStringCastRule.java

```diff
@@ -91,74 +121,98 @@ protected String generateCodeBlockInternal(
         context.declareClassField(
                 className(StringBuilder.class), builderTerm, constructorCall(StringBuilder.class));
 
-        return new CastRuleUtils.CodeWriter()
-                .stmt(methodCall(builderTerm, "setLength", 0))
-                .stmt(methodCall(builderTerm, "append", strLiteral("[")))
-                .forStmt(
-                        methodCall(inputTerm, "size"),
-                        (indexTerm, loopBodyWriter) -> {
-                            String elementTerm = newName("element");
-                            String elementIsNullTerm = newName("elementIsNull");
+        final String resultStringTerm = newName("resultString");
+        final int precision = LogicalTypeChecks.getLength(targetLogicalType);
 
-                            CastCodeBlock codeBlock =
-                                    CastRuleProvider.generateCodeBlock(
-                                            context,
-                                            elementTerm,
-                                            "false",
-                                            // Null check is done at the array access level
-                                            innerInputType.copy(false),
-                                            targetLogicalType);
+        return CharacterFamilyTrimmingAndPaddingCastRule.appendAndTrimStringIfNeeded(
```

Comment:
Because the logical step of doing the append and trim comes after the rest of the code here, can you assign the writer to a variable and invoke append and trim later?

Reply:
done, thx!